wfe/web-front-end.go

```diff
@@ -190,8 +194,22 @@ func (wfe *WebFrontEndImpl) Handler() http.Handler {
 	wfe.NewCert = wfe.BaseURL + NewCertPath
 	wfe.CertBase = wfe.BaseURL + CertPath
 
+	// Only generate directory once
```

Comment:
Is there a chance we could refactor this Handler function? All of these setters seem to be what lead to this Marshal having to be called here. It would be nice if the Handler didn't return an error, the DirectoryPath handler could be a struct that is on the WebFrontEndImpl with the json marshalling being done to create it somewhere else, and then all of this is nice and immutable.